### ISSUE TITLE
bug: Wrong typehint fix

### ISSUE THREAD

**Issue by nchj** [Instagram/Fixit#389]
When I define a class named `List`, and use python buildin `list` as type hint, fixit will wrongly fix `list` to `List`

It seems that fixit is mistaking self defined List for Typing.List

```
class List:
    def __init__(self, data: list):
        self.data = data
```
fixit will fix it to 
```
class List:
    def __init__(self, data: List):
        self.data = data
```

**Comment by llllvvuu** [comment]
How to reproduce?

```python
(fixit) ; cat > a.py        
class List:
    def __init__(self, data: list):
        self.data = data
(fixit) ; fixit fix a.py         
🧼 1 file clean 🧼
(fixit) ; 
```

**Comment by nchj** [comment]
> How to reproduce?
> 
> ```python
> (fixit) ; cat > a.py        
> class List:
>     def __init__(self, data: list):
>         self.data = data
> (fixit) ; fixit fix a.py         
> 🧼 1 file clean 🧼
> (fixit) ; 
> ```

![图片](https://github.com/Instagram/Fixit/assets/28175310/a8cedcf4-c236-4150-a113-cd3974262aea)

**Comment by llllvvuu** [comment]
It must be because of Python version, I am on 3.11.4 so I don't see it:

https://github.com/Instagram/Fixit/blob/84378a845746e2633882010061b808f67f836b3d/src/fixit/rules/use_types_from_typing.py#L25-L31

But indeed it is a bug that it doesn't auto-import (well, that's a bit tricky to implement since you'd have to `import as`; maybe an easier fix would be to return no autofix)

**Comment by nchj** [comment]
cool,&nbsp;thanks，&nbsp;I&nbsp;am&nbsp;using&nbsp;python3.9


---Original---
From: ***@***.***&gt;
Date: Thu, Sep 7, 2023 12:28 PM
To: ***@***.***&gt;;
Cc: &quot;Jia ***@***.******@***.***&gt;;
Subject: Re: [Instagram/Fixit] bug: Wrong typehint fix (Issue #389)




 
It must be because of Python version, I am on 3.11.4 so I don't see it:
 
https://github.com/Instagram/Fixit/blob/84378a845746e2633882010061b808f67f836b3d/src/fixit/rules/use_types_from_typing.py#L25-L31
 
But indeed it is a bug that it doesn't auto-import (well, that's a bit tricky to implement since you'd have to import as; maybe an easier fix would be to return no autofix)
 
&mdash;

**Comment by amyreese** [comment]
Unfortunately, having rules add imports for newly referenced objects is still something that we need to improve.

For this specific case, it would probably be easiest to disable this specific rule in your fixit config:

```toml
disable = ["fixit.rules:UseTypesFromTyping"]
```

See the config guide for details: https://fixit.readthedocs.io/en/stable/guide.html#configuration
